solr/core/src/test/org/apache/solr/cloud/FullSolrCloudDistribCmdsTest.java

```diff
@@ -298,6 +303,29 @@ public void testDeleteByIdCompositeRouterWithRouterField() throws Exception {
         docCountsAfrica.decrementAndGet();
       }
       checkShardCounts.run();
+
+      // Tests for distributing delete by id when route is missing from the request
+
+      { // Send a delete request with no route to shard1 for document on shard2, should be distributed
```

Comment:
This is a "unique" technique, that I haven't seen before in our code base.   I can't really evaluate if it's good or not...   Do we do this elsewhere in the Solr code base?

Reply:
In this case I'm copying the style of the existing test I'm modifying. Code blocks outside of flow-of-control are just for variable scoping, so you can include `final UpdateRequest deleteRequest = new UpdateRequest();` repeatedly.

Reply:
awesome!  I mis-read the patch file!